### ISSUE TITLE
[FR] Resize Uploaded Images before Transforming

### ISSUE THREAD

**Issue by davidhellmann** [nystudio107/craft3-imageoptimize#64]
Maybe a good solution would be that before start with Transforming was to transform the root file to a pre defined size. In a perfect world the client resize the image before the upload but in the real world it doesn't work.

The Client has an 6000x4000px Image to upload.
It's not optimized and the size is sure 5-10mb

now I create a lot of images with this root file is massive server load.
It would be nice to have a setting to define the long side like: 2000px and the root file was downsized to this size. It would be ok that he create this as separate Image to not downsize the root file itself.

## Timeline:

**Upload:** 
mountains.jpg | 6000x4000px

**Create root transform Image:**
mountains_root.jpg | 2000x1333px

**Create alle the sizes**
I think that was a good solution to reduce server load.

**Comment by khalwat** [comment]
Sounds like a decent FR; for now I'd suggest Image Resizer:

https://github.com/verbb/image-resizer

**Comment by davidhellmann** [comment]
Yes use it for now but I'm not sure which one is triggered first :D 
But would be nice to have it in one plugin. 

Sounds good, thanks Andrew!

**Comment by khalwat** [comment]
Image Resizer would be triggered first. ImageOptimize doesn't do anything at all until you create a transform of some kind.
